modules/angular-meteor-collection.js

```diff
@@ -0,0 +1,372 @@
+
+        collection.remove(id, function(err) {
+          if (err)
+            deffered.reject(err);
+          else
+            deffered.resolve({_id: id, action: "removed"});
+
+          $rootScope.$apply();
+        });
+      }
+
+      else {
+        deffered.reject("key can't be null");
+      }
+
+      return deffered.promise;
+    };
+
+    // Function applies async to combine multiple operations (savings, deletions etc)
+    // in one processing.
+    AngularMeteorCollection.sync = function() {
+      var self = this;
+
+      if (this._syncPromise) {
+        $timeout.cancel(this._syncPromise);
+        this._syncPromise = null;
+      }
+
+      // Clearing the watch is needed so no updates are sent to server
+      // while handling updates from the server.
+      if (!this.UPDATING_FROM_SERVER) {
+        this.UPDATING_FROM_SERVER = true;
+        if (!$rootScope.$$phase) $rootScope.$apply();
+      }
+
+      this._syncPromise = $timeout(function () {
+        // Saves changes happened within the previous update from server.
+        collectionUtils.updateCollection(self, self._serverBackup, self.diffArrayFunc);
+        self.UPDATING_FROM_SERVER = false;
+        $rootScope.$apply();
+      }, 0, false);
+    };
+
+    return AngularMeteorCollection;
+}]);
+
+angularMeteorCollection.factory('$meteorCollectionFS', ['$meteorCollection', 'diffArray', function ($meteorCollection, diffArray) {
+  function $meteorCollectionFS(reactiveFunc, auto, collection) {
+    var noNestedDiffArray = function (lastSeqArray, seqArray, callbacks) {
+      return diffArray(lastSeqArray, seqArray, callbacks, true);
+    };
+
+    return new $meteorCollection(reactiveFunc, auto, collection, noNestedDiffArray);
+  }
+
+  return $meteorCollectionFS;
+}]);
+
+angularMeteorCollection.factory('$meteorCollection', ['AngularMeteorCollection', '$rootScope', 'diffArray',
+  function (AngularMeteorCollection, $rootScope, diffArray) {
+    function $meteorCollection(reactiveFunc, auto, collection, diffArrayFunc) {
+      // Validate parameters
+      if (!reactiveFunc) {
+        throw new TypeError("The first argument of $meteorCollection is undefined.");
+      }
+
+      if (!(typeof reactiveFunc == "function" || angular.isFunction(reactiveFunc.find))) {
+        throw new TypeError("The first argument of $meteorCollection must be a function or a have a find function property.");
+      }
+
+      if (!(typeof reactiveFunc == "function")) {
+        var cursorFunc = reactiveFunc.find;
+        var originalCollection = reactiveFunc;
+        collection = angular.isDefined(collection) ? collection : reactiveFunc;
+
+        reactiveFunc = function () {
+          return cursorFunc.apply(originalCollection, [{}]);
+        };
+      }
+
+      var ngCollection = new AngularMeteorCollection(reactiveFunc(), collection, diffArrayFunc);
+      ngCollection._auto = auto !== false;
+      ngCollection.diffArrayFunc = diffArrayFunc || diffArray;
```

Comment:
When changing to your diff package, notice that in some cases (collectionFS) we don't do nested diffs